manifests/osfamily/redhat.pp

```diff
@@ -43,89 +22,124 @@
-  }
+    if $::puppet_agent::manage_pki_dir == true {
+      file { ['/etc/pki', '/etc/pki/rpm-gpg']:
+        ensure => directory,
+      }
+    }
 
-  file { $gpg_path:
-    ensure => present,
-    owner  => 0,
-    group  => 0,
-    mode   => '0644',
-    source => "puppet:///modules/puppet_agent/${keyname}",
-  }
+    file { $legacy_gpg_path:
+      ensure => present,
+      owner  => 0,
+      group  => 0,
+      mode   => '0644',
+      source => "puppet:///modules/puppet_agent/${legacy_keyname}",
+    }
 
-  # Given the path to a key, see if it is imported, if not, import it
-  $gpg_pubkey = "gpg-pubkey-$(echo $(${gpg_cmd} --with-colons ${gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
-  exec {  "import-${keyname}":
-    path      => '/bin:/usr/bin:/sbin:/usr/sbin',
-    command   => "rpm --import ${gpg_path}",
-    unless    => "rpm -q ${gpg_pubkey}",
-    require   => File[$gpg_path],
-    logoutput => 'on_failure',
-  }
+    # Given the path to a key, see if it is imported, if not, import it
+    $legacy_gpg_pubkey = "gpg-pubkey-$(echo $(${gpg_cmd} --with-colons ${legacy_gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
+
+    exec {  "import-${legacy_keyname}":
+      path      => '/bin:/usr/bin:/sbin:/usr/sbin',
+      command   => "rpm --import ${legacy_gpg_path}",
+      unless    => "rpm -q ${legacy_gpg_pubkey}",
+      require   => File[$legacy_gpg_path],
+      logoutput => 'on_failure',
+    }
+
+    file { $gpg_path:
+      ensure => present,
+      owner  => 0,
+      group  => 0,
+      mode   => '0644',
+      source => "puppet:///modules/puppet_agent/${keyname}",
+    }
 
-  if $::puppet_agent::manage_repo == true {
-    $_proxy = $::puppet_agent::disable_proxy ? {
-      true    => '_none_',
-      default => undef,
+    # Given the path to a key, see if it is imported, if not, import it
+    $gpg_pubkey = "gpg-pubkey-$(echo $(${gpg_cmd} --with-colons ${gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
+    exec {  "import-${keyname}":
+      path      => '/bin:/usr/bin:/sbin:/usr/sbin',
+      command   => "rpm --import ${gpg_path}",
+      unless    => "rpm -q ${gpg_pubkey}",
+      require   => File[$gpg_path],
+      logoutput => 'on_failure',
     }
-    yumrepo { 'pc_repo':
-      baseurl             => $source,
-      descr               => "Puppet Labs ${::puppet_agent::collection} Repository",
-      enabled             => true,
-      gpgcheck            => '1',
-      gpgkey              => "${gpg_keys}",
-      proxy               => $_proxy,
-      sslcacert           => $_sslcacert_path,
-      sslclientcert       => $_sslclientcert_path,
-      sslclientkey        => $_sslclientkey_path,
-      skip_if_unavailable => $::puppet_agent::skip_if_unavailable,
+
+    if $::puppet_agent::manage_repo == true {
```

Comment:
For RHEL, the module will setup all of the necessary gpg keys to add to the repo, but the `manage_repo` if statement is way at the bottom and only encapsulates the `yumrepo` resource. If effectively does all of the necessary setup to manage the repo except for the very last step.

Is there a reason to setup all of the GPG keys if we are not managing the repo?? Not sure.